packages/app-page-builder-elements/src/renderers/carousel.tsx

```diff
@@ -0,0 +1,105 @@
+import React from "react";
+import styled from "@emotion/styled";
+import { Elements } from "~/components/Elements";
+import { createRenderer } from "~/createRenderer";
+import { useRenderer } from "~/hooks/useRenderer";
+
+const initializeSwiper = async () => {
+    if (typeof window === "undefined") {
+        return;
+    }
+
+    const register = await import("swiper/element/bundle");
+
+    register.register();
+};
+
+const registerCarousel = () => {
+    new Promise<void>(async resolve => {
+        await initializeSwiper();
+        resolve();
+    });
+};
```

Comment:
Let's simplify this and simply put 

```ts
 if (typeof window === "undefined") {
        return;
    }

    const register = await import("swiper/element/bundle");

    register.register();
```

into `useEffect` that only executes on the initial component render.